src/providers/connectedClusters.ts

```diff
@@ -0,0 +1,159 @@
+import * as k8s from '@kubernetes/client-node';
+import * as path  from 'path';
+import * as vscode from 'vscode';
+
+type CustomViewType = ConnectedCluster | OcmResourceDefinition| OcmResource;
+
+export class ConnectedCluster extends vscode.TreeItem {
+	readonly cluster: k8s.Cluster;
+
+	constructor(cluster: k8s.Cluster) {
+		super(cluster.name, vscode.TreeItemCollapsibleState.Collapsed);
+		this.cluster = cluster;
+		this.tooltip = cluster.server;
+	}
+
+	iconPath = {
+		light: path.join(__dirname, '..', '..', '..', 'images', 'light', 'k8s.svg'),
+		dark: path.join(__dirname, '..', '..', '..', 'images', 'dark', 'k8s.svg'),
+	};
+}
+
+export class OcmResourceDefinition extends vscode.TreeItem {
+	readonly resourceSpec: k8s.V1CustomResourceDefinitionSpec;
+
+	constructor(spec: k8s.V1CustomResourceDefinitionSpec) {
+		super(spec.names.kind, vscode.TreeItemCollapsibleState.Collapsed);
+		this.resourceSpec = spec;
+		this.tooltip = spec.group;
+	}
+
+	iconPath = {
+		light: path.join(__dirname, '..', '..', '..', 'images', 'light', 'ocm.svg'),
+		dark: path.join(__dirname, '..', '..', '..', 'images', 'dark', 'ocm.svg'),
+	};
+}
+
+export class OcmResource extends vscode.TreeItem {
+	constructor(name: string, namespace: string, version: string) {
+		super(name, vscode.TreeItemCollapsibleState.None);
+		this.tooltip = version;
+		this.description = namespace;
+	}
+}
+
+export class ConnectedClustersProvider implements vscode.TreeDataProvider<CustomViewType> {
+	private kubeConfig = new k8s.KubeConfig();
+	private connectedClusters: ConnectedCluster[] = [];
+	private configRefreshedEmitter: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+	readonly onDidChangeTreeData: vscode.Event<void> = this.configRefreshedEmitter.event;
+
+	constructor() {
+		this.updateK8SConfig();
+	}
+
+	refresh(): void {
+		this.updateK8SConfig();
+		this.configRefreshedEmitter.fire();
+	}
```

Comment:
fire?????

Reply:
this fires an event informing vscode on the data being refreshed.

https://code.visualstudio.com/api/references/vscode-api#EventEmitter%3CT%3E